### PR TITLE
Add an index on task_events.task_id.

### PR DESCRIPTION
**Pull request by brianestlin**
TL;DR: In the task history code, `task_events` is often queried with `task_id` as the filter column. In our deployment using an RDS Postgres instance for the task history db, this change reduced `/api/add_task` calls from 200ms to 15ms.

Background:

We at Remind recently noticed that most requests to the central scheduler's `/api/add_task` endpoint were taking 200ms. This seemed wrong. An initial profile showed that most of that time was spent in db_task_history.py. Logging SQLAlchemy queries with timing yielded:

```
2016-03-31 00:04:49,301 [sqlalchemy.engine.base.Engine] INFO: SELECT task_events.id AS task_events_id, task_events.task_id AS task_events_task_id, task_events.event_name AS task_events_event_name, task_events.ts AS task_events_ts 
FROM task_events 
WHERE %(param_1)s = task_events.task_id ORDER BY task_events.ts DESC, task_events.id DESC
2016-03-31 00:04:49,302 [sqlalchemy.engine.base.Engine] INFO: {'param_1': 134441}
2016-03-31 00:04:49,506 [sqlalchemy.timing] DEBUG: Query Time: 0.205163
```

i.e., filtering the `task_events` table by `task_id`, which did not have an index, and was therefore taking 200ms.

### CODE REVIEW

**Comment by Tarrasch** [comment]
Cool! Awesome result! Is the task_history_db working out for you other than this?

This PR is changing a module I don't know so much about. Can you do a `git blame` and look for other contributors who have edited those files and @-mention them to summon them as reviewers of this patch? Once others +1. I can help press "merge".

**Comment by stephenpascoe** [comment]
+1

I think I introduced this in 82812f81. `_find_or_create_task()` started filtering on task_id but I failed to add the index to `TaskEvent` -- I believe it is automatic on MySQL, which we use.

**Comment by Tarrasch** [comment]
Ah, nice to see you're lurking along with all the PRs @stephenpascoe ^^

**Comment by brianestlin** [comment]
@Tarrasch , the task history ui (i.e. `/history` endpoint) has been unusably slow for us, but I think this PR has greatly improved matters there as well. At first glance I think there's some opportunity for usability improvements. :) I hope to take a closer look soon.
